server/controllers/assessmentDayDataController/index.js

```diff
@@ -17,35 +18,53 @@ const AssessmentDayDataController = {
         assessment,
         subject,
       }
-      const storedAssessmentData = await AssessmentDayDataModel.all(
+      const storeDataCursor = await AssessmentDayDataModel.all(
         dataDb,
         collection
       )
-      const assessmentDataService = new AssessmentDayDataService(
-        subject_assessments,
-        storedAssessmentData
-      )
-      const newAssessmentData = assessmentDataService.filterNewData()
-      const updatedData = assessmentDataService.filterUpdatedData()
-
-      await ToCModel.upsert(dataDb, query, metadata)
+      const dataStream = storeDataCursor.stream()
 
-      if (newAssessmentData.length)
-        await AssessmentDayDataModel.saveMany(
-          dataDb,
-          collection,
-          newAssessmentData
+      dataStream.on('data', (doc) => {
+        const { day } = doc
+        const dayDataIndex = subject_assessments.findIndex(
+          (data) => data.day === day
         )
-      for await (const participantData of updatedData) {
-        const { day } = participantData
-
-        await AssessmentDayDataModel.update(
-          dataDb,
-          collection,
-          { day },
-          participantData
+        if (dayDataIndex !== -1) {
+          const removedDocument = subject_assessments.splice(dayDataIndex, 1)
```

Comment:
could you use `find` instead of `findIndex` to get the document so you don't need to splice here?

Reply:
I want to splice it and that's why I am using the index. This way when we reach the on end event after the stream is completed the remaining documents all get written at once, but the updated subjects are iterated and their documents updated.